src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,74 @@
+package org.opentripplanner.middleware.utils;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.Leg;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class ItineraryUtilsTest {
+    @ParameterizedTest
+    @MethodSource("createItineraryCanBeMonitoredCases")
+    public void testItineraryCanBeMonitored(Itinerary itinerary, boolean expectedResult, String message) {
+        Assertions.assertEquals(expectedResult, ItineraryUtils.itineraryCanBeMonitored(itinerary), message);
+    }
+
+    private static Stream<Arguments> createItineraryCanBeMonitoredCases() {
+        Leg transitLeg = new Leg();
+        transitLeg.mode = "BUS";
+        transitLeg.transitLeg = true;
+
+        Leg rentalBikeLeg = new Leg();
+        rentalBikeLeg.mode = "BICYCLE_RENT";
+        rentalBikeLeg.rentedBike = true;
+
+        Leg rentalCarLeg = new Leg();
+        rentalCarLeg.mode = "CAR_RENT";
+        rentalCarLeg.rentedCar = true;
+
+        Leg rentalMicromobilityLeg = new Leg();
+        rentalMicromobilityLeg.mode = "MICROMOBILITY_RENT";
+        rentalMicromobilityLeg.rentedVehicle = true;
+
+        Leg walkLeg = new Leg();
+        walkLeg.mode = "WALK";
+
+        Leg rideHailLeg = new Leg();
+        rideHailLeg.mode = "CAR_HAIL";
+        rideHailLeg.hailedCar = true;
+
+        Itinerary blankItinerary = new Itinerary();
+
+        Itinerary itineraryWithTransitNoRentals = new Itinerary();
+        itineraryWithTransitNoRentals.legs = List.of(transitLeg, walkLeg);
+
+        Itinerary itineraryWithRentalBikeWithoutTransit = new Itinerary();
+        itineraryWithRentalBikeWithoutTransit.legs = List.of(walkLeg, rentalBikeLeg);
+
+        Itinerary itineraryWithTransitAndRentalBike = new Itinerary();
+        itineraryWithTransitAndRentalBike.legs = List.of(walkLeg, transitLeg, rentalBikeLeg);
+
+        Itinerary itineraryWithTransitAndRentalCar = new Itinerary();
+        itineraryWithTransitAndRentalCar.legs = List.of(walkLeg, transitLeg, rentalCarLeg);
+
+        Itinerary itineraryWithTransitAndRentalMicromobility = new Itinerary();
+        itineraryWithTransitAndRentalMicromobility.legs = List.of(walkLeg, transitLeg, rentalMicromobilityLeg);
+
+        Itinerary itineraryWithTransitAndRideHail = new Itinerary();
+        itineraryWithTransitAndRideHail.legs = List.of(walkLeg, transitLeg, rideHailLeg);
+
+        return Stream.of(
```

Comment:
This does seem simpler than creating a test case class, but it might result in runtime errors if an incorrect number or type of arguments are entered. I'm fine with keeping this as is for now and it's good to know this is a possible way of creating arguments for a parameterized test.